docs/core/tutorials/unloadability-howto.md

```diff
@@ -0,0 +1,359 @@
+
+The .NET Core 3.0 brings support of unloadability using `AssemblyLoadContext`. You can now load a set of assemblies into an `AssemblyLoadContext`, execute methods in them or just inspect them using reflection and finally unload the `AssemblyLoadContext`. That results in unloading of the assemblies loaded into that `AssemblyLoadContext`.
+
+There is one noteworthy difference between the unloading using `AssemblyLoadContext` and using AppDomains. With AppDomains, the unloading is forced. At the unload time, all threads running in the target AppDomain are aborted, managed COM objects created in the AppDomain that's being unloaded are destroyed, etc. 
+
+With the new way that uses `AssemblyLoadContext`, the unload is "cooperative". Calling the `Unload` method on the `AssemblyLoadContext` just initiates the unloading. But the unload will not complete until there are no threads having code from the assemblies loaded into the `AssemblyLoadContext` on their call stacks and until there are no strong references to types from the assemblies loaded into the `AssemblyLoadContext`, their instances and the assemblies themselves. And that means that GC needs to collect those first.
+## Using unloadable AssemblyLoadContext
+You can find a complete sample [at this link](https://github.com/dotnet/samples/tree/master/core/tutorials/Unloading). This section contains step by step detailed tutorial on a simplest way how to load a .NET Core application into an unloadable AssemblyLoadContext, execute its entry point and then unload it.
+### How to create an unloadable AssemblyLoadContext
+You need to derive your own class from the `AssemblyLoadContext` and overload its Load method. That method is used to resolve references to all assemblies that are dependencies of assemblies loaded into that `AssemblyLoadContext`. 
+Here is an example of the simplest custom `AssemblyLoadContext`:
+```C#
+class TestAssemblyLoadContext : AssemblyLoadContext
+{
+    public TestAssemblyLoadContext() : base(isCollectible: true)
+    {
+    }
+
+    protected override Assembly Load(AssemblyName name)
+    {
+        return null;
+    }
+}
+```
+As you can see, the `Load` method returns `null`. That means that all the dependency assemblies are loaded into the default context and only the assemblies explicitly loaded into the new context are in this context. 
+
+In case you want to load some or all of the dependencies into the `AssemblyLoadContext` too, you can use the `AssemblyDependencyResolver` in the `Load` method. The `AssemblyDependencyResolver` resolves the assembly names to absolute assembly file paths using the *.deps.json file contained in the directory of the main assembly loaded into the context and using assembly files in that directory.
```

Comment:
```suggestion
In case you want to load some or all of the dependencies into the `AssemblyLoadContext` too, you can use the `AssemblyDependencyResolver` in the `Load` method. The `AssemblyDependencyResolver` resolves the assembly names to absolute assembly file paths using the `*.deps.json` file contained in the directory of the main assembly loaded into the context and using assembly files in that directory.
```

Reply:
@jkoritzinsky added a plugin sample for this. Reference it?

Reply:
Do you mean a sample that uses *.deps.json in some interesting way or a just a sample that uses AssemblyDependencyResolver? If it is the latter, I already have a link above to my own unloadability sample that uses it.

Reply:
My sample is also in that official sample repo. I've merged it in two days ago. It covers unloadability and since I wanted to make it a boiler plate that people can use, I've used the AssemblyDependencyResolver there too. If you've missed the link in the text above, it is here:
https://github.com/dotnet/samples/tree/master/core/tutorials/Unloading

Reply:
In that case you should use a code reference tag to automatically import the code from the sample like the following

`[!code-csharp[Assembly-load-context-with-assembly-dependency-resolver](~/samples/blob/master/core/tutorials/Unloading/Host/Program.cs)]`

You might also want to seperate out the ALC into a different file to make the code reference work a little cleaner.

Reply:
I didn't want to include the code from my sample here. I see this doc and the sample as separate entities. This doc focuses on a simple version where the Load method always returns NULL and just mentions the more advanced way briefly so that people know it exist. 
The sample is a separate full blown end to end example of using unloadable ALC that complements this article. I didn't actually know you were working on your sample and frankly even didn't know about the AssemblyDependencyResolver existence when I've completed the first version of my sample and I was originally using just the simple ALC returning NULL from the Load overried. @jkotas then pointed out during review of that sample that it might be nice to make it more complete by adding the usage of the AssemblyDependencyResolver. Thus I've added it there and mentioned it in this doc.

Reply:
That tag is a specially handled by the markdown engine that docs.microsoft.com uses. Github doesn't understand it fully.